### PR TITLE
feat(nginx): opentelemetry-cpp v1.0.0-rc3

### PR DESCRIPTION
**Pull request by johanneswuerbach**
Bump to https://github.com/open-telemetry/opentelemetry-cpp/releases/tag/v1.0.0-rc3

### CODE REVIEW

File: instrumentation/nginx/src/otel_ngx_module.cpp [hunk 1]
```diff
@@ -17,7 +17,7 @@ extern ngx_module_t otel_ngx_module;
 #include "nginx_utils.h"
 #include "propagate.h"
 #include <opentelemetry/context/context.h>
-#include <opentelemetry/exporters/otlp/otlp_exporter.h>
+#include <opentelemetry/exporters/otlp/otlp_grpc_exporter.h>
```
Comment (johanneswuerbach) [inline instrumentation/nginx/src/otel_ngx_module.cpp:20]: This was due to https://github.com/open-telemetry/opentelemetry-cpp/pull/810

**Review by seemk — APPROVED** [review]
Thanks! That was fast :smile:

File: instrumentation/nginx/CMakeLists.txt [hunk 2]
```diff
@@ -44,4 +45,5 @@ target_link_libraries(otel_ngx_module
   ${OPENTELEMETRY_CPP_LIBRARIES}
   gRPC::grpc++
   CURL::libcurl
+  nlohmann_json::nlohmann_json
```
Comment (johanneswuerbach) [inline instrumentation/nginx/CMakeLists.txt:48]: This is coming from https://github.com/open-telemetry/opentelemetry-cpp/pull/810, not sure if we should attempt to drop this as `OTLP/HTTP+JSON exporter` isn't used at the moment.

Comment (seemk) [reply instrumentation/nginx/CMakeLists.txt:48]: Does it build when not using `find_package(nlohmann_json)` and not linking to it?

Comment (johanneswuerbach) [reply instrumentation/nginx/CMakeLists.txt:48]: Nope :-(

Comment (seemk) [reply instrumentation/nginx/CMakeLists.txt:48]: Otel CPP external dependencies are starting to get a bit out of hand, wish it had none and would be distributed as a single library file. :smile: 

Anyway, thanks for the PR!

**Pull request merged**